recipes/libmodbus/all/conanfile.py

```diff
@@ -62,38 +64,25 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
-        env = VirtualBuildEnv(self)
-        env.generate()
-
         tc = AutotoolsToolchain(self)
         if Version(self.version) < "3.1.8":
             tc.configure_args.append("--without-documentation")
         tc.configure_args.append("--disable-tests")
         if is_msvc(self) and check_min_vs(self, "180", raise_invalid=False):
             tc.extra_cflags.append("-FS")
-        env = tc.environment()
-        if is_msvc(self):
-            compile_wrapper = unix_path(self, os.path.join(self.source_folder, "build-aux", "compile"))
-            ar_wrapper = unix_path(self, self.conf.get("user.automake:lib-wrapper", check_type=str))
-            env.define("CC", f"{compile_wrapper} cl -nologo")
-            env.define("CXX", f"{compile_wrapper} cl -nologo")
-            env.define("LD", "link -nologo")
-            env.define("AR", f"{ar_wrapper} \"lib -nologo\"")
-            env.define("NM", "dumpbin -symbols")
-            env.define("OBJDUMP", ":")
-            env.define("RANLIB", ":")
-            env.define("STRIP", ":")
-        tc.generate(env)
```

Comment:
We typically dont accept changes unrelated to the PR - making an exception here, as our observations match yours: in a lot of recipes this whole thing is unnecessary and the build scripts already use aversion of the compile wrapper by default